js/src/product-feed/review-request/index.js

```diff
@@ -29,7 +30,7 @@ const ReviewRequest = ( { account = {} } ) => {
 	if (
 		! mcDataHasFinishedResolution ||
 		! hasFinishedResolution ||
-		! showNotice( accountData.status ) ||
+		! showNotice( accountData?.status ) ||
```

Comment:
Not sure about this tbh. As far as I know data could be `undefined` or `null`. Yes, we are checking hasFinishedResolution to cover that, but I will sleep better if there is not a fatal error crashing the entire feed if there is some kind of malformation in the account. 🙏

Reply:
> As far as I know data could be `undefined` or `null`.

❔ 
Maybe I missed something, so I'm not pretty sure when the data would be `undefined` or `null`. Unless the GET/POST `mc/review` API will respond successfully with other data structures rather than [`AccountStatus`](https://github.com/woocommerce/google-listings-and-ads/blob/255a575a0d5feefba428fdf8dbaa08dc9449aca6/js/src/data/actions.js#L65-L73), otherwise, the `account.data` here is always the `AccountStatus` structure.
- The data structure of `account` prop is the shape of [`useAppSelectDispatch`](https://github.com/woocommerce/google-listings-and-ads/blob/255a575a0d5feefba428fdf8dbaa08dc9449aca6/js/src/hooks/useAppSelectDispatch.js#L28-L36), which is always an object.
- The data structure of `account.data` here is always the [`mc_review_request` data structure](https://github.com/woocommerce/google-listings-and-ads/blob/255a575a0d5feefba428fdf8dbaa08dc9449aca6/js/src/data/reducer.js#L40-L45) provided from the `getMCReviewRequest` selector of wp-data.
- The `get_review_status` function has [handling for the `null` case](https://github.com/woocommerce/google-listings-and-ads/blob/255a575a0d5feefba428fdf8dbaa08dc9449aca6/src/API/Site/Controllers/MerchantCenter/RequestReviewController.php#L235-L239). (It seems not quite accurate that the doc of `get_review_status` says [it may return `null`](https://github.com/woocommerce/google-listings-and-ads/blob/255a575a0d5feefba428fdf8dbaa08dc9449aca6/src/API/Site/Controllers/MerchantCenter/RequestReviewController.php#L229).) And even somehow that function returns `null` to callers, the response data will still be like this:
   ```json
   {
     "status": null,
     "cooldown": null,
     "issues": null,
     "reviewEligibleRegions": null
   }
   ```

Reply:
> As far as I know data could be `undefined` or `null`.

❔ 
Maybe I missed something, so I'm not pretty sure when the data would be `undefined` or `null`. Unless the GET/POST `mc/review` API will respond successfully with other data structures rather than [`AccountStatus`](https://github.com/woocommerce/google-listings-and-ads/blob/255a575a0d5feefba428fdf8dbaa08dc9449aca6/js/src/data/actions.js#L65-L73), otherwise, the `account.data` here is always the `AccountStatus` structure.
- The data structure of `account` prop is the shape of [`useAppSelectDispatch`](https://github.com/woocommerce/google-listings-and-ads/blob/255a575a0d5feefba428fdf8dbaa08dc9449aca6/js/src/hooks/useAppSelectDispatch.js#L28-L36), which is always an object.
- The data structure of `account.data` here is always the [`mc_review_request` data structure](https://github.com/woocommerce/google-listings-and-ads/blob/255a575a0d5feefba428fdf8dbaa08dc9449aca6/js/src/data/reducer.js#L40-L45) provided from the `getMCReviewRequest` selector of wp-data.
- The `get_review_status` function has [handling for the `null` case](https://github.com/woocommerce/google-listings-and-ads/blob/255a575a0d5feefba428fdf8dbaa08dc9449aca6/src/API/Site/Controllers/MerchantCenter/RequestReviewController.php#L235-L239). (It seems not quite accurate that the doc of `get_review_status` says [it may return `null`](https://github.com/woocommerce/google-listings-and-ads/blob/255a575a0d5feefba428fdf8dbaa08dc9449aca6/src/API/Site/Controllers/MerchantCenter/RequestReviewController.php#L229).) And even somehow that function returns `null` to callers, the response data will still be like this:
   ```json
   {
     "status": null,
     "cooldown": null,
     "issues": null,
     "reviewEligibleRegions": null
   }
   ```

Reply:
Right. I tried to log the account payload and is true that is always with the AccountStatus shape. So no undefined. I was kinda sure at some point it was like ` { hasFinishedResolution: false, data: undefined ...}` but seems like it's not the case 🤔